metrics/exporter.go

```diff
@@ -130,6 +131,12 @@ func UpdateExporterFromConfigMapWithOpts(opts ExporterOptions, logger *zap.Sugar
 // to prevent a race condition between reading the current configuration
 // and updating the current exporter.
 func UpdateExporter(ops ExporterOptions, logger *zap.SugaredLogger) error {
+	// We don't check if ops.secrets is `nil` right now, because this is a transition step.
```

Comment:
Any open issue to track this? Are we going to do it before we migrate to OpenTelemetry? If not, I'd prefer not add the comments for the part after `Eventually`.

Also createMetricsConfig seems the better place for this comment.

Reply:
I filed #1273 and linked it here. I know we currently do all the ExporterOptions checks in createMetricsConfig, but I think they are more appropriate in exporter.go where ExporterOptions is actually defined. Might as well catch the error as early as possible in the call stack.

The new UpdateExporterFromConfigMapWithOpts does this.